doc/pac/Port Access Control.md

```diff
@@ -0,0 +1,844 @@
+
+
+### 3.4.2 mabd
+mabd provides the MAC Authentication Bypass (MAB) functionality. MAB is intended to provide 802.1x unaware clients controlled access to the network using the devices’ MAC address as an identifier. This requires that the known and allowable MAC address and corresponding access rights be pre-populated in the authentication server.  
+
+PAC supported authentication methods for MAB are as given below:
+- CHAP
+- EAP-MD5
+- PAP
+
+### 3.4.3 hostapd
+Hostapd is an open source implementation of 802.1x standard and the Linux application is supplied with wpa_suplicant package. The wired driver module of hostapd is adapted to communicate with pacd via socket interface
+
+. hostapd gets its configuration from the hostapd.conf file generated by hostapdmgrd.
+
+### 3.4.4 hostapdmgrd
+hostapdmgr reads hostapd specific configuration from SONiC DBs and populates the hostapd.conf. It further notifies the hostapd to re-read the configuration file.
+
+### 3.4.5 Interaction between modules
+
+*hostapd(802.1X)*   
+
+hostapd comes to know of an 802.1x client attempting authentication via an EAP exchange. It informs pacd of this client by conveying the client MAC. If the authentication method selected by pacd is 802.1X, pacd sends an event to hostapd for authenticating the user. The client is however authenticated via MAB If the authentication method selected here is MAB.
+
+hostapd informs pacd about the result of the authentication. hostapd also passes all the authorization parameters it receives from the RADIUS Server to the pacd. These are used for configuring the switch to allow authenticated client traffic.
+
+*mabd(MAB)*   
+
+When user or client tries to authenticate and the method selected is MAB, the pacd sends an event to mabd for authenticating the user. The client’s MAC address is sent to mabd for the same. 
+
+pacd learns client’s MAC address through an hardware rule to copy-to-CPU the packets from unknown source MAC addresses.
+
+mabd informs pacd about the result of the authentication. mabd also passes all the authorization parameters it receives from the RADIUS Server to the pacd. These are used for configuring the NAS to allow authenticated client traffic.
+
+
+## 3.5 SyncD
+
+No specific changes are needed in syncd for PAC.
+
+## 3.6 SAI
+
+Existing SAI attributes are used for this implementation and there is no new SAI attribute requirement.
+
+### 3.6.1 Host interface traps
+
+Leveraged **SAI_HOSTIF_TRAP_TYPE_EAPOL** to trap EAP packets (Ethertype - 0x888E) to the CPU.   
+
+
+### 3.6.2 Bridge port learning modes
+PAC uses the following bridge port learning modes to drop/trap all unknown source MAC packets.   
+- SAI_BRIDGE_PORT_FDB_LEARNING_MODE_DROP
+- SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW
+- SAI_BRIDGE_PORT_FDB_LEARNING_MODE_CPU_TRAP
+
+
+
+### 3.6.3 FDB
+PAC uses **SAI_FDB_ENTRY_ATTR_PACKET_ACTION** with **SAI_PACKET_ACTION_DROP** to put the static FDB entry in discard state.   
```

Comment:
@rck-innovium Whether Both Innovium and Prestera (AC5x) line support src-discard per FDB entry ?